k8-cluster/on-prem/rke2/ansible/README.md

```diff
@@ -0,0 +1,142 @@
+# RKE2 Kubernetes Cluster Setup using Ansible
+
+This repository contains an Ansible playbook and associated configurations to set up an RKE2 Kubernetes cluster with a primary control plane, subsequent control plane nodes, agent nodes, and an etcd cluster. The playbooks are designed to install RKE2, configure nodes, and ensure the cluster is ready for use.
+
+## Table of Contents
+- [Prerequisites](#prerequisites)
+- [Inventory Setup](#inventory-setup)
+- [Playbooks](#playbooks)
+- [Templates](#templates)
+- [Kubernetes setup via Ansible playbook](#kubernetes-setup-via-Ansible-playbook)
+- [Configuration Details](#configuration-details)
+
+## Prerequisites
+
+Before running the playbooks, ensure the following:
+
+- Install Ansible on your local machine.
+- Ensure that nodes are accessible from your local machine for SSH authentication.
+- Verify that the SSH private key for the nodes is available on your local machine.
+- Supported operating systems include Ubuntu 24.04 and other Debian-based distributions.
+
+
+## Inventory Setup
+
+The `hosts.ini` file contains the configuration of your nodes. It is structured as follows:
+
+```
+# Primary control plane node. This is the main node where RKE2 will be installed first.
```

Comment:
Dont display the contents of hosts.ini.
Rather make hosts.ini self explanatery.

Add comment to define the default roles for each type.